swiftui/ChartboostMediationDemo/Ads/Fullscreen/FullscreenAdController.swift

```diff
@@ -0,0 +1,156 @@
+
+//
+//  InterstitialAdController.swift
+//  ChartboostMediationDemo
+//
+//  Copyright © 2023 Chartboost. All rights reserved.
+//
+
+import UIKit
+import ChartboostMediationSDK
+
+/// A basic implementation of a controller for Chartboost Mediation interstitial ads.  It is capable of loading and showing a full screen interstitial ad
+/// for a single placement.  This controller is also its own `CHBHeliumInterstitialAdDelegate` so that it is in full control
+/// of the ad's lifecycle.
+class FullscreenAdController: NSObject, ObservableObject {
+    /// The entry point for the Chartboost Mediation SDK.
+    private let chartboostMediation = Helium.shared()
+
+    /// The placement that this controller is for.
+    private let placementName: String
+
+    /// An instance of the fullscreen ad that this class controls the lifecycle of when using the new API.
+    private var fullscreenAd: ChartboostMediationFullscreenAd?
+
+    /// A state for demo purposes only so that long activity processes can be communicated to a view.
+    @Published private(set) var activityState: ActivityState = .idle
+
+    /// Initializer for the view model.
+    /// - Parameter placementName: Placement to use.
+    init(placementName: String) {
+        self.placementName = placementName
+    }
+
+    /// Load the interstitial ad.
+    /// - Parameter keywords: Optional keywords that can be associated with the advertisement placement.
+    func load(keywords: HeliumKeywords? = nil) {
+        // Attempt to load the ad only if it has not already been created and requested to load.
+        guard fullscreenAd == nil else {
+            print("[Warning] interstitial advertisement has already been loaded")
+            return
+        }
+
+        // Notify the demo UI
+        activityState = .running
+
+        // loadFullscreenAd expects keywords to be `[String : String]` instead of `HeliumKeywords?`.
+        let keywords = keywords?.dictionary ?? [:]
+        let request = ChartboostMediationAdLoadRequest(placement: placementName, keywords: keywords)
+        // Load the fullscreen ad, which will make a request to the network. Upon completion, a
+        // ChartboostMediationFullscreenAdLoadResult will be passed to the completion block.
+        chartboostMediation.loadFullscreenAd(with: request) { [weak self] result in
+            guard let self = self else { return }
+            if let ad = result.ad {
+                ad.delegate = self
+                self.fullscreenAd = ad
+
+                log(action: "load", placementName: placementName, error: result.error)
+                // Notify the demo UI
+                activityState = .idle
+            } else {
+                fullscreenAd = nil
+                self.log(action: "Load error", placementName: placementName, error: result.error)
+
+                // `.failed` requires a non-optional error type
+                let error = result.error ?? NSError(domain: "com.chartboost.mediation.demo", code: 0)
```

Comment:
```suggestion
                let error = result.error
```

If the error object in the `.failed` state is optional, we don't need the default error here.